From 927e9e5707ff67887efd83c65ba36b7b81f12ce6 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 19 Aug 2018 13:39:04 -0500
Subject: [PATCH 1/6] Typecast password to const char * explicitly (py3)

src/mysql_capi.c:1215:10: error: no matching function for call to
'mysql_real_connect'
    res= mysql_real_connect(&self->session, host, user, password, database,
         ^~~~~~~~~~~~~~~~~~
/xxx/include/mysql.h:417:18: note: candidate function not viable: no known
conversion from 'PyObject *' (aka '_object *') to 'const char *' for 4th
argument
MYSQL *         STDCALL mysql_real_connect(MYSQL *mysql, const char *host,
---
 mysql-connector-python/src/mysql_capi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mysql-connector-python/src/mysql_capi.c b/mysql-connector-python/src/mysql_capi.c
index d2e5bf2..506f2ac 100644
--- a/mysql-connector-python/src/mysql_capi.c
+++ b/mysql-connector-python/src/mysql_capi.c
@@ -1449,7 +1449,7 @@ MySQL_connect(MySQL *self, PyObject *args, PyObject *kwds)
 #endif
 
     Py_BEGIN_ALLOW_THREADS
-    res = mysql_real_connect(&self->session, host, user, password,
+    res = mysql_real_connect(&self->session, host, user, (const char *)password,
                              database, port, unix_socket, client_flags);
     Py_END_ALLOW_THREADS
 
-- 
2.45.2

